pkg/home/data.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The envd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package home
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/cockroachdb/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type dataManager interface {
+	InitDataDir(name string) (string, error)
+}
+
+func (m *generalManager) InitDataDir(name string) (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get user home dir")
+	}
+	newDataDir := filepath.Join(home, ".envd", "data", name)
```

Comment:
I think we should avoid ~/.envd, maybe we keep it in ~/.cache/envd?

Reply:
The semantic here I think is different from cache. Once a dataset is created, it is not expected to be deleted when clearing the cache. Also cache means it can be rebuilt from some other way, but I didn't see how we can rebuild user's data folder, as this is just an arbitrary folder user can play with